code/modules/mob/living/carbon/carbon.dm

```diff
@@ -745,9 +745,13 @@
 		. += INFINITY
 
 // permeability: now slightly more sane and probably functional!
-/mob/living/carbon/get_permeability(def_zone)
+/mob/living/carbon/get_permeability(def_zone, linear = FALSE)
 	if(def_zone)
-		var/permeability = 1 / (2**(getarmor(def_zone, BIO) / 15))
+		if(!can_inject(def_zone))
+			return 0
+		if(getarmor(def_zone, BIO) >= 100)
+			return 0
+		var/permeability = linear ? (100 - getarmor(def_zone)) / 100 : 1 / (2**(getarmor(def_zone, BIO) / 15))
```

Comment:
Uh.. does this work?
``checkarmor()``, called in ``getarmor()`` on Human, will early return if ``type`` is not provided but ``def_zone`` is, so linear shouldn't be working here, no?